cmd/cosign/cli/sign/sign_blob.go

```diff
@@ -98,18 +99,15 @@ func SignBlobCmd(ro *options.RootOptions, ko options.KeyOpts, regOpts options.Re
 		signedPayload.Bundle = cbundle.EntryToBundle(entry)
 	}
 
-	// if bundle is specified, just do that and ignore the rest
-	if ko.RFC3161TimestampPath != "" {
-		signedPayload.Base64Signature = base64.StdEncoding.EncodeToString(sig)
-
-		contents, err := json.Marshal(signedPayload)
+	if ko.RFC3161TimestampPath != "" && rfc3161Timestamp != nil {
```

Comment:
Should we handle the case where I ask for the timestamp to be written, but I don't actually get a timestamp?

Feels worth a warning at least.

Reply:
It seems like this drops the TSA timestamp when there's no timestamp path specified? it seems like there should be a default one then -- otherwise I'd think to only set TSAServerURL (and then my timestamp disappears)

Reply:
We already do that if you don't specify `--output-signature` or `--output-certificate` 🤷 Maybe a `y/N` prompt would be the right amount of annoying here, you could bypass with `--yes`

I'd also be comfortable just waiting until we cut over to the new bundle format and then making `--bundle` required.

Reply:
@znewman01 I think the signature is always output to STDOUT though regardless, so you can always pass it.

I guess it's fair that the certificate is swallowed when you don't specify a output-certificate or bundle even if you use keyless, that seems error-prone to me too.

Reply:
(But also the cert is written to STDERR at some point during the command output)

Reply:
I'm going to require that you set both TSAServerURL and RFC3161TimestampPath, to always write the timestamp. If we add in writing to Rekor at some point, we can change the requirement.